### ISSUE TITLE
🐛 index out of bounds: the len is 247 but the index is 1197

### ISSUE THREAD

**Issue by evan-taylor** [biomejs/biome#7455]
### Environment information

```block
CLI:
  Version:                      2.2.2
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.3.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/11.4.2

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Workspace:
  Open Documents:               0
```

### What happened?

We installed biome via. the latest version of ultracite in our codebase. When we attempted to check or fix, we get this error, that appears at least one time, sometimes a few times:

```Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_module_graph/src/js_module_info/module_resolver.rs:439:33
Thread Name: biome::workspace_worker_2
Message: index out of bounds: the len is 247 but the index is 1197

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_module_graph/src/js_module_info/module_resolver.rs:439:33
Thread Name: biome::workspace_worker_10
Message: index out of bounds: the len is 247 but the index is 1197```

### Expected result

It should not throw an error, and fix or check appropriately. 

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
